syntax = "proto3";

package app;

service Reddit {

  rpc SendPost (PostRequest) returns (stream GetPost) {}
}

message PostRequest {
  string response = 1;
}

message GetPost {
  string id = 1;
  string title = 2;
  string score = 3;
  string author = 4;
  string author_flair_text = 5;
  string removed_by = 6;
  string total_awards_received = 7;
  string awarders = 8;
  string created_utc = 9;
  string full_link = 10;
  string num_comments = 11;
  string over_18 = 12;
}
